File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Java/src/main/java/Ficheros/BurguerPig/models/dto/IngredienteDTO.java
package Ficheros.BurguerPig.models.dto;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.io.Serializable;
import java.util.List;

@Root(name = "ingrediente")
public class IngredienteDTO implements Serializable {

    @Attribute(name = "name")
    private String name;

    @Attribute(name = "precio")
    private String price;

    @Attribute(name = "id")
    private String id;

    public IngredienteDTO() {
        // Constructor vacío sin argumentos
    }

    public IngredienteDTO(String name, String price, String id) {
        this.name = name;
        this.price = price;
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public String getPrice() {
        return price;
    }

    public String getId() {
        return id;
    }
}

File: buerguer-pig/JCZhang-BurguerPigFicheros-Kotlin/src/main/kotlin/dto/IngredienteDTO.kt
package dto


import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "Ingrediente")
data class IngredienteDTO(

    @field:Attribute(name = "id")
    @param:Attribute(name = "id")
    var id: Int,

    @field:Element(name = "nombre")
    @param:Element(name = "nombre")
    val nombre: String,

    @field:Attribute(name = "precio")
    @param:Attribute(name = "precio")
    val precio: Double
)


@Root(name="ingredientes")
data class IngredientesListDto(
    @field: ElementList(name = "ingredientes", inline= true)
    @param: ElementList(name = "ingredientes", inline= true)
    val ingredientes: List<IngredienteDTO>
)